model-optimizer/extensions/ops/sparse_reshape.py

```diff
@@ -28,26 +30,41 @@ def supported_attrs(self):
 
     @staticmethod
     def infer(node: Node):
+        name = node.soft_get('name', node.id)
+
         input_indices_shape = node.in_port(0).data.get_shape()
         input_indices_value = node.in_port(0).data.get_value()
-        input_shape_value = node.in_port(1).data.get_value()
-        new_shape_value = node.in_port(2).data.get_value()
+        input_shape = node.in_port(1).data.get_value()
+        new_shape = node.in_port(2).data.get_value()
         new_shape_shape = node.in_port(2).data.get_shape()
 
-        assert input_shape_value is not None and new_shape_value is not None, \
+        assert input_shape is not None and new_shape is not None, \
             "Values for input shape and new shape must be defined"
-        assert np.count_nonzero(new_shape_value == -1) <= 1, \
+        assert len(np.argwhere(new_shape == -1)) <= 1, \
             "Value -1 occurs in new shape value more than once"
+        assert len(np.argwhere(new_shape < -1)) == 0, \
+            "Only non-negative or -1 values are allowed"
+
+        output_shape = np.ma.masked_array(new_shape, mask=new_shape == -1)
```

Comment:
in this case you have different `fill_value` that marks dynamic dimension. Please correct